File: ThirtyDaysOfCodeInJava/class_vs_instance_4.java
/*
Class vs Instance - https://www.hackerrank.com/challenges/30-class-vs-instance?h_r=next-challenge&h_v=zen

Task 
Write a Person class with an instance variable, , and a constructor that takes an integer, , as a parameter. The constructor must assign  to  after confirming the argument passed as is not negative; if a negative argument is passed as , the constructor should set  to  and print Age is not valid, setting age to 0.. In addition, you must write the following instance methods:

yearPasses() should increase the  instance variable by .
amIOld() should perform the following conditional actions:
If , print You are young..
If  and , print You are a teenager..
Otherwise, print You are old..
*/
import java.io.*;
import java.util.*;

class Person{
	int age;

	public Person(int age){
		if(age >= 0){
			this.age = age;
		}else{
			System.out.println("Age is not valid, setting age to 0.");
			this.age = 0;
		}
	}

	public void amIOld(){
		if(this.age < 13)
			System.out.println("You are young.");
		else if(this.age >= 13 && this.age < 18)
			System.out.println("You are a teenager.");
		else
			System.out.println("You are old.");
	}

	public void yearPasses() {
	        this.age++;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for (int i = 0; i < T; i++) {
			int age = sc.nextInt();
			Person p = new Person(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			System.out.println();
        }
		sc.close();
    }

}

File: ThirtyDaysOfCodeInPython/class_vs_instance_4.py
# Class vs Instance - https://www.hackerrank.com/challenges/30-class-vs-instance?h_r=next-challenge&h_v=zen

# Task 
# Write a Person class with an instance variable, , and a constructor that takes an integer, , as a parameter. The constructor must assign  to  after confirming the argument passed as is not negative; if a negative argument is passed as , the constructor should set  to  and print Age is not valid, setting age to 0.. In addition, you must write the following instance methods:

# yearPasses() should increase the  instance variable by 3
# amIOld() should perform the following conditional actions:
# If , print You are young..
# If  and , print You are a teenager..
# Otherwise, print You are old..
import sys


class Person():
	age = 0

	def __init__(self,age):
		if(age >= 0):
			self.age = age
		else:
			print "Age is not valid, setting age to 0."
			self.age = 0

	def amIOld(self):
		if(self.age < 13):
			print "You are young."
		elif(18 > self.age >= 13):
			print "You are a teenager."
		else:
			print "You are old."

	def yearPasses(self):
		self.age+=1

scan = sys.stdin
print "waiting for input"
t = int(scan.readline())
i=0
j=0

while (i < t):
	age = int(scan.readline())
	p = Person(age)
	p.amIOld()
	while(j < 3):
		p.yearPasses()
		j+=1
	p.amIOld()
	i+=1

